src/Microsoft.Sbom.Api/Utils/PackageDetailsFactory.cs

```diff
@@ -0,0 +1,256 @@
+            }
+            else
+            {
+                pomLocation += $"/{componentNamespaceParts[i]}";
+            }
+        }
+
+        pomLocation += $"/{componentName}/{componentVersion}/{componentName}-{componentVersion}.pom";
+
+        // Check for file permissions on the .m2 directory before attempting to check if the file exists
+        if (fileSystemUtils.DirectoryHasReadPermissions(mavenPackagesPath))
+        {
+            if (fileSystemUtils.FileExists(pomLocation))
+            {
+                return pomLocation;
+            }
+        }
+
+        return null;
+    }
+
+    // Takes in a scanned component and attempts to find the associated nuspec file. If it is not found then it returns null.
+    private string GetNuspecLocation(ScannedComponent scannedComponent)
+    {
+        var nuspecLocation = string.Empty;
+        var nugetPackagesPath = SettingsUtility.GetGlobalPackagesFolder(new NullSettings());
+
+        var componentName = scannedComponent.Component.PackageUrl?.Name.ToLower();
+        var componentVersion = scannedComponent.Component.PackageUrl?.Version;
+        var componentType = scannedComponent.Component.PackageUrl?.Type.ToLower();
+
+        if (componentType == "nuget")
+        {
+            nuspecLocation = $"{nugetPackagesPath}/{componentName}/{componentVersion}/{componentName}.nuspec";
+        }
+
+        // Check for file permissions on the .nuget directory before attempting to check if every file exists
+        if (fileSystemUtils.DirectoryHasReadPermissions(nugetPackagesPath))
+        {
+            if (fileSystemUtils.FileExists(nuspecLocation))
+            {
+                return nuspecLocation;
+            }
+        }
+
+        return null;
+    }
+
+    // Get the developers section of the pom file
+    private (string, string, PackageDetailsObject) ParsePom(string pomLocation)
+    {
+        var pomInfo = new PackageDetailsObject();
+
+        try
+        {
+            var pomBytes = File.ReadAllBytes(pomLocation);
+            using var pomStream = new MemoryStream(pomBytes, false);
+
+            var doc = new XmlDocument();
+            doc.Load(pomStream);
+
+            XmlNode developersNode = doc["project"]?["developers"];
+            XmlNode licensesNode = doc["project"]?["licenses"];
+
+            var name = doc["project"]?["artifactId"]?.InnerText;
+            var version = doc["project"]?["version"]?.InnerText;
+
+            if (developersNode != null)
+            {
+                foreach (XmlNode developerNode in developersNode.ChildNodes)
+                {
+                    var developerName = developerNode["name"]?.InnerText;
+                    var developerEmail = developerNode["email"]?.InnerText;
+
+                    if (!string.IsNullOrEmpty(developerName))
+                    {
+                        pomInfo.Supplier += $"{developerName}";
+                    }
+
+                    if (!string.IsNullOrEmpty(developerEmail))
```

Comment:
nit: I guess if developer name will be empty, then supplier field will have a bit weird format: ` (email@example.com) `. Is it okay?